src/code/z_bg_item.c

```diff
@@ -1,99 +1,138 @@
 #include "global.h"
 
-void BcCheck3_BgActorInit(DynaPolyActor* actor, UNK_TYPE4 param_2) {
-    actor->bgId = -1;
-    actor->unk148 = 0;
-    actor->unk14C = 0;
-    actor->unk154 = param_2;
-    actor->unk_158 = 0;
+#define DYNAPOLY_STATE_NONE 0
+#define DYNAPOLY_STATE_RIDING_FALLING 1 << 0
+#define DYNAPOLY_STATE_RIDING_MOVING 1 << 1
+#define DYNAPOLY_STATE_RIDING_ROTATING 1 << 2
+#define DYNAPOLY_STATE_SWITCH_PRESSED 1 << 3
+#define DYNAPOLY_STATE_HEAVY_SWITCH_PRESSED 1 << 4
+
+void DynaPoly_Init(DynaPolyActor* dynaActor, s32 flags) {
```

Comment:
In oot we know some things about the values `flags` takes, see the `DynaPolyMoveFlag` enum. Maybe you want to wait until more is confirmed though?

Reply:
I'm confident in the use cases for these in both OOT and MM and find the names to be better. I didn't see really anything referring to enemies for example.

Reply:
I'm actually not sure what DynaPolyMoveFlag does in OoT. When decomping bgcheck in oot I simply used the existing enumeration since it makes modifications easier to do later on.

Reply:
I don't see where values for the `flags` field have been named, only `stateFlags`, could you point to where if you have? And if you have, existing usages of `DynaPoly_Init` would need their second arg updating to use the defines/enum.

Reply:
Ahh I misunderstood, my bad. Yeah I didn't look into what the flags meant at all because it's not really used by the file. Figured it was out of scope for this documentation.